### PR TITLE
Phils laboratory

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
This pr is for review only, please wait for me to go over it with Ryan before closing

Adds alternative solution for most popular items function with test. 

If we decide to go with my changes I will make a different branch to make sure no merge conflicts are dealt with appropriately.

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -25,21 +25,8 @@ def no_orders?
     item_orders.empty?
   end
 
-  def total_bought
-    orders.sum(:quantity)
+  def self.most_popular_item
+    order_quant = Item.joins(:item_orders).select("items.*, sum(quantity) as total_ordered").group(:id).order("total_ordered")
   end
```
Comment (ryandbarnett) [inline app/models/item.rb:30]: Is this being invoked anywhere?

File: app/models/item_order.rb [hunk 2]
```diff
@@ -7,4 +7,12 @@ class ItemOrder <ApplicationRecord
...
     price * quantity
   end
+
+  def self.most_popular_items
+    items_hash = ItemOrder.group(:item_id).sum(:quantity).sort_by {|item_id, quantity| -quantity}
+    result = items_hash.map do |item_id, quantity|
+      [Item.find(item_id), quantity]
+    end
+    result[0..4]
+  end
```
Comment (ryandbarnett) [inline app/models/item_order.rb:17]: Alright I think I understand what this does. Nice work

**Review by ryandbarnett — APPROVED** [review]

**Comment by ryandbarnett** [comment]
Don't merge! But what you have is good!

**Comment by dhalverson** [comment]
Closing, review only

**Pull request closed without merging**